src/test/scala/uk/gov/nationalarchives/tdr/api/routes/TransfersAgreementRouteSpec.scala

```diff
@@ -90,6 +97,57 @@ class TransfersAgreementRouteSpec extends AnyFlatSpec with Matchers with TestReq
     response.errors.head.message should equal(expectedResponse.errors.head.message)
   }
 
+  "The api" should "return an existing transfer agreement for a user owned consignment" in {
+    val consignmentSql = s"insert into consignmentapi.Consignment (SeriesId, UserId) VALUES (1,'$userId')"
+    val sql = "INSERT INTO consignmentapi.TransferAgreement (ConsignmentId, AllPublicRecords, AllCrownCopyright, " +
+      "AllEnglish, AllDigital, AppraisalSelectionSignedOff, SensitivityReviewSignedOff, TransferAgreementId) " +
+      "VALUES (1, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE);"
+    val psConsignment: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(consignmentSql)
+    val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(sql)
+    ps.executeUpdate()
+    psConsignment.executeUpdate()
+
+    val expectedResponse: GraphqlQueryData = expectedQueryResponse("data_all")
+    val response: GraphqlQueryData = runTestQuery("query_alldata", validUserToken())
+    response.data should equal(expectedResponse.data)
+  }
+
+  "The api" should "return no transfer agreement if it doesn't exist" in {
+    val consignentSql = s"insert into consignmentapi.Consignment (SeriesId, UserId) VALUES (1,'$userId')"
+    val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(consignentSql)
+    ps.executeUpdate()
+    ps.executeUpdate()
+
+    val expectedResponse: GraphqlQueryData = expectedQueryResponse("data_none")
+    val response: GraphqlQueryData = runTestQuery("query_alldata", validUserToken())
+    response.data should equal(expectedResponse.data)
+  }
+
+  "The api" should "return an error if the consignment id isn't provided" in {
+    val consignentSql = s"insert into consignmentapi.Consignment (SeriesId, UserId) VALUES (1,'$userId')"
+    val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(consignentSql)
+    ps.executeUpdate()
+    ps.executeUpdate()
+
+    val expectedResponse: GraphqlQueryData = expectedQueryResponse("data_consignmentidmissing")
+    val response: GraphqlQueryData = runTestQuery("query_missingconsignmentid", validUserToken())
+    response.errors.length should equal(1)
+    response.errors.head.message should equal(expectedResponse.errors.head.message)
+  }
+
+  "The api" should "return an error if the user doesn't own the consignment" in {
+    val otherUserId = UUID.randomUUID().toString
+    val consignentSql = s"insert into consignmentapi.Consignment (SeriesId, UserId) VALUES (1,'$otherUserId')"
+    val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(consignentSql)
+    ps.executeUpdate()
+    ps.executeUpdate()
+
+    val expectedResponse: GraphqlQueryData = expectedQueryResponse("data_invalidconsignment")
```

Comment:
Should this be some kind of authorization error rather than "invalidconsignment"?

Reply:
I've changed the name to data_notowner because this is what we've used before and it makes more sense.

Reply:
I might revisit this with the authorisation work so we can consolidate the auth errors, if that's OK?